### PR TITLE
awc: improve module documentation

### PR DESCRIPTION
**Pull request by popzxc**
## PR Type

Docs

## PR Checklist
Check your PR fulfills the following:

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [ ] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview

Subjectively, one great part of `reqwest` is its documentation, which provides usage examples for most common users' needs right on the main page of `docs.rs`.

This PR extends the `awc main doc-comment with the similar list of examples, so that users can see how to use library without digging deep into the crate docs (or even the source code).

### CODE REVIEW

**Comment by popzxc** [comment]
CI fails due to error in `actix-http`:

```
error[E0432]: unresolved import `actix_tls::SslError`
##[error]   --> actix-http/src/service.rs:198:46
    |
198 |     use actix_tls::{openssl::HandshakeError, SslError};
    |                                              ^^^^^^^^ no `SslError` in the root

error[E0432]: unresolved import `actix_tls::SslError`
##[error]   --> actix-http/src/h1/service.rs:101:46
    |
101 |     use actix_tls::{openssl::HandshakeError, SslError};
    |                                              ^^^^^^^^ no `SslError` in the root

error[E0432]: unresolved import `actix_tls::SslError`
##[error]   --> actix-http/src/h2/service.rs:100:46
    |
100 |     use actix_tls::{openssl::HandshakeError, SslError};
    |                                              ^^^^^^^^ no `SslError` in the root

error: aborting due to 3 previous errors
```

Not sure if I should do something about it `¯\_(ツ)_/¯`

**Comment by robjtede** [comment]
Nope CI failures are not your fault. Just need to rebase onto #1655 when it's merged.

File: awc/src/lib.rs [hunk 1]
```diff
@@ -4,24 +4,98 @@
     clippy::borrow_interior_mutable_const,
     clippy::needless_doctest_main
 )]
-//! An HTTP Client
+
+//! `awc` is a HTTP/WebSocket library implemented atop of `actix` stack.
```
Comment (Neopallium) [inline awc/src/lib.rs]: 'is a HTTP/WebSocket Client library..."

**Review by Neopallium — APPROVED** [review]

File: awc/src/lib.rs [hunk 2]
```diff
@@ -4,24 +4,98 @@
     clippy::borrow_interior_mutable_const,
     clippy::needless_doctest_main
 )]
-//! An HTTP Client
+
+//! `awc` is a HTTP/WebSocket Client library implemented atop of `actix` stack.
```
Comment (robjtede) [inline awc/src/lib.rs]: Don't want to suggest is is using the `actix` crate.

```suggestion
//! `awc` is a HTTP and WebSocket client library built using the Actix ecosystem.
```

File: awc/src/lib.rs [hunk 3]
```diff
@@ -51,7 +125,9 @@ pub use self::sender::SendClientRequest;
 
 use self::connect::{Connect, ConnectorWrapper};
 
-/// An HTTP Client
+/// An asynchronous web Client.
```
Comment (robjtede) [inline awc/src/lib.rs]:
```suggestion
/// An asynchronous HTTP and WebSocket client.
```

File: awc/src/lib.rs [hunk 4]
```diff
@@ -4,24 +4,98 @@
...
+
+//! `awc` is a HTTP/WebSocket Client library implemented atop of `actix` stack.
 //!
-//! ```rust
-//! use actix_rt::System;
-//! use awc::Client;
+//! ## Making a GET request
 //!
+//! ```rust
+//! # use actix_rt::System;
```
Comment (robjtede) [inline awc/src/lib.rs]: None of these imports should be needed.

```suggestion
```

File: awc/src/lib.rs [hunk 5]
```diff
@@ -4,24 +4,98 @@
...
+//! ### JSON
+//!
+//! ```rust
+//! # use std::collections::HashMap;
+//! # use actix_rt::System;
+//! # #[actix_rt::main]
+//! # async fn main() -> Result<(), awc::error::SendRequestError> {
+//! let mut map = HashMap::new();
+//! map.insert("lang", "rust");
+//! map.insert("body", "json");
```
Comment (robjtede) [inline awc/src/lib.rs]: would prefer the use of the serde_json::json! construction.

File: awc/src/lib.rs [hunk 6]
```diff
@@ -4,24 +4,98 @@
...
+//! ## Making a GET request
 //!
+//! ```rust
+//! # use actix_rt::System;
 //! #[actix_rt::main]
-//! async fn main() {
-//!    let mut client = Client::default();
+//! # async fn main() -> Result<(), awc::error::SendRequestError> {
+//! let mut client = awc::Client::default();
+//! let response = client.get("http://www.rust-lang.org") // <- Create request builder
```
Comment (robjtede) [inline awc/src/lib.rs:16]: if this can work with minimal change as the https url then would prefer that

**Review by robjtede — CHANGES_REQUESTED** [review]

File: awc/src/lib.rs [hunk 7]
```diff
@@ -59,17 +56,16 @@
...
 //!
 //! ```rust
-//! # use std::collections::HashMap;
-//! # use actix_rt::System;
 //! # #[actix_rt::main]
 //! # async fn main() -> Result<(), awc::error::SendRequestError> {
-//! let mut map = HashMap::new();
-//! map.insert("lang", "rust");
-//! map.insert("body", "json");
+//! let mut request = serde_json::json!({
```
Comment (robjtede) [inline awc/src/lib.rs]:
```suggestion
//! let request = serde_json::json!({
```

**Review by robjtede — CHANGES_REQUESTED** [review]

**Review by robjtede — APPROVED** [review]
nice job, thanks 👍

**Comment by popzxc** [comment]
Ah, Windows tests fail the same way as in #1654.

I'll restore URLs to HTTP and open an issue.

**Edit:** Actually, it's already reported: https://github.com/actix/actix-web/issues/1560

**Comment by robjtede** [comment]
> I'll restore URLs to HTTP

Fair enough. We'll merge with this and I shall try to remember to change them back to HTTPS before stable release. We should get this working on Windows.

**Pull request merged**